conformance_tests/sysman/test_sysman_power/src/test_sysman_power.cpp

```diff
@@ -852,34 +856,37 @@ TEST_F(
         power_sustained_Initial = power_limits_descriptors[i];
         power_sustained_Max.limit = std::numeric_limits<int>::max();
         power_sustained_Initial.limit *= 2;
-      }
-      if (power_sustained_Max.limitValueLocked == false) {
-        lzt::set_power_limits_ext(power_card_handle, &count_power,
-                                  &power_sustained_Max);
-        auto power_limits_descriptors_getMax =
-            lzt::get_power_limits_ext(power_card_handle, &count_power);
-        for (auto p_power_limits_descriptor_get :
-             power_limits_descriptors_getMax) {
-          if (p_power_limits_descriptor_get.level ==
-              ZES_POWER_LEVEL_SUSTAINED) {
-            power_sustained_getMax = p_power_limits_descriptor_get;
+
+        if (power_sustained_Max.limitValueLocked == false) {
+          lzt::set_power_limits_ext(power_card_handle, &count_power,
+                                    &power_sustained_Max);
+          auto power_limits_descriptors_getMax =
+              lzt::get_power_limits_ext(power_card_handle, &count_power);
+          for (auto p_power_limits_descriptor_get :
+               power_limits_descriptors_getMax) {
+            if (p_power_limits_descriptor_get.level ==
+                ZES_POWER_LEVEL_SUSTAINED) {
+              power_sustained_getMax = p_power_limits_descriptor_get;
+            }
           }
-        }
-        lzt::set_power_limits_ext(power_card_handle, &count_power,
-                                  &power_sustained_Initial);
-        auto power_limits_descriptors_get =
-            lzt::get_power_limits_ext(power_card_handle, &count_power);
-        for (auto p_power_limits_descriptor_get :
-             power_limits_descriptors_get) {
-          if (p_power_limits_descriptor_get.level ==
-              ZES_POWER_LEVEL_SUSTAINED) {
-            power_sustained_get = p_power_limits_descriptor_get;
+          lzt::set_power_limits_ext(power_card_handle, &count_power,
+                                    &power_sustained_Initial);
+          auto power_limits_descriptors_get =
+              lzt::get_power_limits_ext(power_card_handle, &count_power);
+          for (auto p_power_limits_descriptor_get :
+               power_limits_descriptors_get) {
+            if (p_power_limits_descriptor_get.level ==
+                ZES_POWER_LEVEL_SUSTAINED) {
+              power_sustained_get = p_power_limits_descriptor_get;
+            }
           }
+          EXPECT_LE(power_sustained_get.limit, power_sustained_getMax.limit);
+        } else {
+          LOG_INFO << "Set limit not supported due to sustained "
+                      "limitValueLocked flag is true";
         }
-        EXPECT_LE(power_sustained_get.limit, power_sustained_getMax.limit);
       } else {
-        LOG_INFO << "Set limit not supported due to sustained "
-                    "limitValueLocked flag is true";
+          LOG_INFO << "Sustained power limit not supported";
```

Comment:
I think this info can be outside for loop, after iterating over all descriptors

Reply:
done